lib/adapter.js

```diff
@@ -2062,17 +2062,17 @@ function Adapter(options) {
                         defState = oldObj.common.def;
                     }
                     if (defState !== undefined) {
-                        let currentState;
+                        let currentStateObj;
                         try {
-                            currentState = await this.getStateAsync(id);
-                        } catch (e) {
-                            currentState = null;
+                            currentStateObj = await this.getForeignStateAsync(id);
+                        } catch {
+                            // do nothing
                         }
-                        if (!currentState) {
+                        if (!currentStateObj) {
```

Comment:
due the fact that Dominic  and I both mixed it up with state value, it may be a better name now to avoid confusion in the future